### PR TITLE
[Python] Add lzma module

### LINKED ISSUE #19817: [Python] - lzma missing
Test Add-on: [lzma.test.zip](https://github.com/xbmc/xbmc/files/6739002/lzma.test.zip)

### Describe the bug
Here is a clear and concise description of what the problem is:

Python3 on Kodi 19 Android fails to load lzma module.
lzma is part of the standard lib since Python 3.3 so should be included
https://docs.python.org/3/library/lzma.html

lzma on Windows loads fine

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

lzma module loads

## Actual Behavior

```
Starting Kodi (19.1 (19.1.0) Git:20210508-85e05228b4). Platform: Android ARM 64-bit
File "/data/user/0/org.xbmc.kodi/cache/apk/assets/python3.8/lib/python3.8/lzma.py", line 27, in <module>
from _lzma import *
ModuleNotFoundError: No module named '_lzma'
```

## Possible Fix

add `lzma` here: https://github.com/fuzzard/xbmc/blob/master/tools/depends/target/Makefile#L95
add `_lzma _lzmamodule.c -I$(prefix)/include -L$(prefix)/lib -llzma` here: https://github.com/fuzzard/xbmc/blob/master/tools/depends/target/python3/modules.setup#L377

Similar solved issue:
bz2 missing https://github.com/xbmc/xbmc/issues/18152
fixed with pr: https://github.com/xbmc/xbmc/pull/18220

### To Reproduce

1. 
2.
3.

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [X] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
Kodi 19.1 Android ARM 64-bit

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by matthuisman**
## Description

fixes: https://github.com/xbmc/xbmc/issues/19817
Build lzma and add as a dependency for Python3

Maybe candidate for back-port to 19.X if wanted?
Just let me know and ill do a backport PR.
Not concerned either way. But may help stop future 19.X issues about lzma not being available

## Motivation and context

LZMA built-in library is missing on non-windows platforms.
lzma is part of the standard lib since Python 3.3 so should be included
https://docs.python.org/3/library/lzma.html

## How has this been tested?
Test plugin: https://github.com/xbmc/xbmc/files/6739002/lzma.test.zip
Android ARM64 Confirmed latest nightly - lzma fails
confirmed kodi compiles and lzma works with this PR.

## What is the effect on users?

lzma works :)

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/xz/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
+include ../../Makefile.include
+DEPS= ../../Makefile.include Makefile
+
+# lib name, version
+LIBNAME=xz
+VERSION=5.2.4
+SOURCE=$(LIBNAME)-$(VERSION)
+ARCHIVE=$(SOURCE).tar.bz2
```
Comment (fuzzard) [inline tools/depends/target/xz/Makefile:8]: Might be best to use gz (just uniformity with other depends really)  and any reason to not use 5.2.5?

Comment (matthuisman) [reply tools/depends/target/xz/Makefile:8]: i basically just use the native xz as a template
https://github.com/xbmc/xbmc/blob/master/tools/depends/native/xz/Makefile
which uses 5.2.4 and .bz2

I'll update to 5.2.5 and .gz package :)

Comment (fuzzard) [reply tools/depends/target/xz/Makefile:8]: Can stay with the bz2 then, do update to the 5.2.5 though

Comment (fuzzard) [reply tools/depends/target/xz/Makefile]: Ping @wsnipex for an upload to mirrors for https://tukaani.org/xz/xz-5.2.5.tar.bz2

Comment (matthuisman) [reply tools/depends/target/xz/Makefile]: you happy with it being called xz?
the native xz actually builds /src/xz/xz
where as this builds /src/liblzma/

Comment (fuzzard) [reply tools/depends/target/xz/Makefile]: Fine with the name as xz

Comment (wsnipex) [reply tools/depends/target/xz/Makefile]: uploaded to mirrors.

Comment (matthuisman) [reply tools/depends/target/xz/Makefile]: not sure why the build failed. i suspect because I force pushed during the initial build?

**Comment by wsnipex** [comment]
jenkins build this please

**Review by wsnipex — APPROVED** [review]

**Comment by matthuisman** [comment]
thanks guys :)
Did we want back-ported to 19.X?
19.x is python3 so a case could be made it should be available there?

**Comment by wsnipex** [comment]
> thanks guys :)
> Did we want back-ported to 19.X?
> 19.x is python3 so a case could be made it should be available there?

up to @DaveTBlake, but if fixes some addons, it might be worth it. Risk should be pretty low

**Pull request merged**